File: Industrial-Programming/pr4/pr4_3.cpp
#include <ctime>
#include <iostream>

int getUTCDifference() {
    std::time_t t = std::time(nullptr);
    std::tm local = *std::localtime(&t);
    std::tm utc = *std::gmtime(&t);

    int localHour = local.tm_hour;
    int utcHour = utc.tm_hour;

    int diff = localHour - utcHour;

    // Если местное время находится в часовом поясе, отличном от UTC,
    // то разница может быть отрицательной. В этом случае мы добавляем 24,
    // чтобы получить положительное число.
    if (diff < 0) {
        diff += 24;
    }

    return diff;
}

int main() {
    std::cout << "Разница между местным временем и UTC составляет " << getUTCDifference() << " часов." << std::endl;
    return 0;
}

File: Industrial-Programming/pr4/pr4_3.js
//3

function getUTCDifference() {
    const now = new Date();
    const localHour = now.getHours();
    const utcHour = now.getUTCHours();

    let diff = localHour - utcHour;

    // If the local time is in a time zone different from UTC,
    // the difference may be negative. In this case, we add 24
    // to get a positive number.
    if (diff < 0) {
        diff += 24;
    }

    return diff;
}

console.log("The difference between local time and UTC is " + getUTCDifference() + " hours.");